ProjectLighthouse.Servers.Website/Pages/UserPage.cshtml.cs

```diff
@@ -29,35 +31,25 @@ public class UserPage : BaseLayout
     public List<SlotEntity>? QueuedSlots;
 
     public UserEntity? ProfileUser;
+
+    public bool CanAccessProfile;
+    public bool CanAccessSlots;
+
     public UserPage(DatabaseContext database) : base(database)
-    {}
+    { }
 
     public async Task<IActionResult> OnGet([FromRoute] int userId)
     {
         this.ProfileUser = await this.Database.Users.FirstOrDefaultAsync(u => u.UserId == userId);
         if (this.ProfileUser == null) return this.NotFound();
 
         // Determine if user can view profile according to profileUser's privacy settings
-        if (this.User == null || !this.User.IsAdmin)
-        {
-            switch (this.ProfileUser.ProfileVisibility)
-            {
-                case PrivacyType.PSN:
-                {
-                    if (this.User != null) return this.NotFound();
-
-                    break;
-                }
-                case PrivacyType.Game:
-                {
-                    if (this.ProfileUser != this.User) return this.NotFound();
-
-                    break;
-                }
-                case PrivacyType.All: break;
-                default: throw new ArgumentOutOfRangeException();
-            }
-        }
+        this.CanAccessProfile = this.ProfileUser.ProfileVisibility.CanAccess(
+            this.User != null, 
+            this.ProfileUser == this.User || this.User != null && this.User.IsModerator);
+        this.CanAccessSlots = this.ProfileUser.LevelVisibility.CanAccess(
+            this.User != null, 
+            this.ProfileUser == this.User || this.User != null && this.User.IsModerator);
```

Comment:
You could extract the authenticated and owner bools so you don't have to calculate them twice. Something like this: <sub>(please don't commit this directly I'm writing this in a browser)</sub>
```suggestion
        bool isLoggedIn = this.User != null;
        bool isOwner = this.ProfileUser == this.User || this.User != null && this.User.IsModerator;
        this.CanAccessProfile = this.ProfileUser.ProfileVisibility.CanAccess(isLoggedIn, isOwner);
        this.CanAccessSlots = this.ProfileUser.LevelVisibility.CanAccess(isLoggedIn, isOwner);
```